Revert upstream commits r12779 and r13045.
We can do this because we fix GNOME #316087 in our GTK+ packages.
Index: nautilus-2.20.0/src/file-manager/fm-list-view.c
===================================================================
--- nautilus-2.20.0.orig/src/file-manager/fm-list-view.c	2007-08-21 11:37:48.000000000 +0200
+++ nautilus-2.20.0/src/file-manager/fm-list-view.c	2007-09-24 11:55:24.918925454 +0200
@@ -69,7 +69,6 @@
 #include <libnautilus-private/nautilus-view-factory.h>
 #include <libnautilus-private/nautilus-clipboard.h>
 #include <libnautilus-private/nautilus-cell-renderer-pixbuf-emblem.h>
-#include <libnautilus-private/nautilus-cell-renderer-text-ellipsized.h>
 
 struct FMListViewDetails {
 	GtkTreeView *tree_view;
@@ -1338,7 +1337,7 @@
 
 	for (l = nautilus_columns; l != NULL; l = l->next) {
 		NautilusColumn *nautilus_column;
-		int column_num;		
+		int column_num, font_size;
 		char *name;
 		char *label;
 		float xalign;
@@ -1361,6 +1360,9 @@
 			view->details->pixbuf_cell = (GtkCellRendererPixbuf *)cell;
 			
 			view->details->file_name_column = gtk_tree_view_column_new ();
+			gtk_tree_view_column_set_expand (view->details->file_name_column, TRUE);
+			font_size = PANGO_PIXELS (pango_font_description_get_size (GTK_WIDGET(view)->style->font_desc));
+			gtk_tree_view_column_set_min_width (view->details->file_name_column, 20*font_size);
 			g_object_ref (view->details->file_name_column);
 			gtk_object_sink (GTK_OBJECT (view->details->file_name_column));
 			view->details->file_name_column_num = column_num;
@@ -1382,7 +1384,11 @@
 							     "pixbuf_emblem", FM_LIST_MODEL_SMALLEST_EMBLEM_COLUMN,
 							     NULL);
 			
-			cell = nautilus_cell_renderer_text_ellipsized_new ();
+			cell = gtk_cell_renderer_text_new ();
+			g_object_set (cell,
+				      "ellipsize", PANGO_ELLIPSIZE_END,
+				      "ellipsize-set", TRUE,
+				      NULL);
 			view->details->file_name_cell = (GtkCellRendererText *)cell;
 			g_signal_connect (cell, "edited", G_CALLBACK (cell_renderer_edited), view);
 			g_signal_connect (cell, "editing-canceled", G_CALLBACK (cell_renderer_editing_canceled), view);
